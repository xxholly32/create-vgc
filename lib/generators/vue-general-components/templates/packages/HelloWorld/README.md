# `HelloWorld`

> TODO: description

## Usage

```
const helloWorld = require('HelloWorld');

// TODO: DEMONSTRATE API
```
