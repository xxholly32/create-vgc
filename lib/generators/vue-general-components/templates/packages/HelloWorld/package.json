{
  "name": "@vgc/hello-world",
  "version": "0.0.1",
  "description": "demo for vue-general-components",
  "keywords": [
    "hello-world",
    "vue-general-components",
    "vgc"
  ],
  "author": "xxholly32 <xxholly32@163.com>",
  "homepage": "",
  "license": "ISC",
  "main": "dist/index.js",
  "directories": {
    "dist": "dist",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "scripts": {
    "test": "jest",
    "build": "rollup --config ./build/rollup.build.js"
  },
  "devDependencies": {
    "@rollup/plugin-node-resolve": "^7.1.3",
    "babel-core": "^6.26.0",
    "babel-jest": "^26.0.1",
    "babel-preset-env": "^1.6.0",
    "jest": "^26.0.1",
    "jest-serializer-vue": "^0.2.0",
    "rollup": "^1.16.7",
    "rollup-plugin-babel": "^3.0.0",
    "rollup-plugin-commonjs": "^10.0.1",
    "rollup-plugin-vue": "^5.0.1",
    "vue": "^2.6.11",
    "vue-jest": "^3.0.5",
    "vue-test-utils": "^1.0.0-beta.11"
  },
  "jest": {
    "moduleFileExtensions": [
      "js",
      "vue"
    ],
    "moduleNameMapper": {
      "^@/(.*)$": "<rootDir>/src/$1"
    },
    "transform": {
      "^.+\\.js$": "<rootDir>/node_modules/babel-jest",
      ".*\\.(vue)$": "<rootDir>/node_modules/vue-jest"
    },
    "snapshotSerializers": [
      "<rootDir>/node_modules/jest-serializer-vue"
    ]
  },
  "dependencies": {
    "@rollup/plugin-json": "^4.0.3"
  }
}
