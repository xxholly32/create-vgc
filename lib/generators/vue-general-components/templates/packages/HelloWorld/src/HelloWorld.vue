<template>
  <div>
    <p class="demo-box">This is VGC hello {{ message }} demo</p>
  </div>
</template>

<script>
export default {
  name: "HelloWorld",
  data() {
    return {
      message: "world",
    };
  },
};
</script>

<style>
/* css */
.demo-box {
  font-size: 24px;
  font-weight: bold;
}
</style>
