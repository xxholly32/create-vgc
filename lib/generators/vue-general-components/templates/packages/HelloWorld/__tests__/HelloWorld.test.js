import Vue from "vue";
import HelloWorld from "../src/HelloWorld";

describe("HelloWorld.test.js", () => {
  let cmp, vm;

  beforeEach(() => {
    cmp = Vue.extend(HelloWorld); // Create a copy of the original component
    vm = new cmp({
      data: {
        // Replace data value with this fake data
        message: "world",
      },
    }).$mount(); // Instances and mounts the component
  });

  it('message includes "world"', () => {
    expect(vm.$el.querySelector("p").outerHTML).toContain("world");
  });

  it("has the expected html structure", () => {
    expect(vm.$el).toMatchSnapshot();
  });
});
