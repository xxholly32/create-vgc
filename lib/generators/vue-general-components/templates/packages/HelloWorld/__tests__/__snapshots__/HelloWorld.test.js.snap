// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.test.js has the expected html structure 1`] = `
<div>
  <p
    class="demo-box"
  >
    This is VGC hello world demo
  </p>
</div>
`;
